### PR TITLE
Remove todo and validate RFile prefix

### PR DESCRIPTION
**Pull request by cshannon**
Leaving this as a draft PR for now as it needs work because tests break due to the new validation. I need to go through and see what other issues there might be from this PR before it is merged when I have time.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/metadata/ValidationUtil.java [hunk 1]
```diff
@@ -52,8 +56,13 @@ public static Path validate(Path path) {
...
 
   public static void validateRFileName(String fileName) {
-    Objects.requireNonNull(fileName);
-    // TODO: In 3.0.0 validate that filename starts with FilePrefix
+    Preconditions.checkArgument(!StringUtils.isEmpty(fileName),
+        "Provided filename is null or empty.");
+
+    // This will make sure the prefix is valid and throw an IllegalArgumentException if the prefix
+    // does not match a known value
+    var unused = FilePrefix.fromPrefix(fileName.substring(0, 1));
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/metadata/ValidationUtil.java:64]: Naming the variable "unused" doesn't actually prevent some tools from reporting that it is unused. Could do some kind of assertion check here (`if (FilePrefix.fromPrefix(...) == null) { throw .... ; }`). Depending on what it returns, this check could probably be combined with the subsequent `if` statement that verifies it the file name `.endsWith` the expected stuff.

Or just add a trivial `void` method in `FilePrefix` called `validatePrefix` that works similarly to `fromPrefix`, but doesn't have an expected return statement (and maybe does the substring for you).

Comment (cshannon) [reply core/src/main/java/org/apache/accumulo/core/metadata/ValidationUtil.java:64]: I had put that in there as I've seen (I think) SpotBugs complain in the past but can definitely clean that up when I get some more time to work on this.

**Comment by ctubbsii** [comment]
Need to be sure this validation method only checks prefixes for filenames that we've stored... and this code path doesn't get executed when a user provides their own file names as in put (rfile-info or as a file that is being bulk imported, for example), because users can name their RFiles anything they like.

I'm also concerned generally about doing this... for a long time, when HDFS or another issue caused an RFile to be corrupt or missing, we have recommended users perform metadata surgery to place an RFile directly. I can imagine other scenarios where a user has done that for surgical/maintenance reasons (like maybe manually compacting some files that is inconvenient for them to compact using our built-in compactors), and may have their own naming convention for the file they place (and for which they add an entry to the metadata table).

This validation could break that situation where users have done that.... and not for any great reason. These file names are just conventions... they aren't a strict requirement. By default, Accumulo should work regardless of the file name, and these names would only matter for custom compaction strategies, trash policies, etc. that the user chose to deploy.

Maybe a warning could be logged for unexpected name prefixes instead of a hard failure?

**Comment by EdColeman** [comment]
The reference in the documentation for creating "empty" files is under the Advanced Troubleshooting section - Data Corruption (https://accumulo.apache.org/docs/2.x/troubleshooting/advanced).  (It was easiest to have an `empty.rf` file lying around and just use that)

**Comment by ctubbsii** [comment]
> The reference in the documentation for creating "empty" files is under the Advanced Troubleshooting section - Data Corruption (https://accumulo.apache.org/docs/2.x/troubleshooting/advanced). (It was easiest to have an `empty.rf` file lying around and just use that)

In that example, the file is usually renamed to whatever the missing/corrupt file was. But I'm imagining cases where a user might perform surgery and not use our internal naming conventions (they shouldn't have to). It would be a surprising breakage.

**Comment by cshannon** [comment]
Before I continue on with any more updates should I go ahead and close this for now? The validation breaks some tests and @ctubbsii voiced concerns about doing this. Based on all the test failures I'm seeing and the issues @ctubbsii mentioned i'm hesitant to make this change now.

**Comment by ctubbsii** [comment]
> Before I continue on with any more updates should I go ahead and close this for now? The validation breaks some tests and @ctubbsii voiced concerns about doing this. Based on all the test failures I'm seeing and the issues @ctubbsii mentioned i'm hesitant to make this change now.

I'm slightly opposed to this change (`-0`) for the reasons I mentioned, but I'm more interested in why those tests are failing. It seems this method might be expecting to receive full paths to files, not just the file part, as this PR had assumed. It's either that, or those tests are providing bad test cases, that this PR unintentionally caught.

**Comment by cshannon** [comment]
I can rerun and check the tests (it may just be the tests that need fixing) I haven't looked to closely into it yet but I don't think there's an issue with the actual code itself other than tests using weird names

**Comment by cshannon** [comment]
@ctubbsii - So a good example of a test that breaks is `RFileClientTest`. This is simply due to the test itself. The test just creates temp files that don't match the expected prefix name, so it's just a test issue that would need to be fixed. That was the first one that broke, I'm not sure how many other tests create temp files that wouldn't validate without running the full suite but I'm going to go ahead and close this anyways I think.

The todo i guess could be removed in a separate small commit if you want as I know we didn't want to leave around TODOs.

**Comment by cshannon** [comment]
Closing for now based on the feedback in this PR, we can always re-open later.

**Pull request closed without merging**